cert-manager/letsencrypt-staging.yaml

```diff
@@ -8,5 +8,34 @@ spec:
     server: https://acme-staging-v02.api.letsencrypt.org/directory
     privateKeySecretRef:
       name: letencrypt-staging-account-key
-    # Enable the HTTP01 challenge mechanism for this Issuer
+
+    # Leave the old-format HTTP01 config *enabled* until we have confirmed that
+    # all Certificate resources no longer specify a spec.acme field.
     http01: {}
+
+    solvers:
+
+    # Make a special exception for gcsweb - it has its own Ingress resource, so
+    # define a dedicated solver for it.
+    - selector:
+        dnsNames:
```

Comment:
Notice that here we explicitly specify `dnsNames` that should use this solver, whereas in the other solver entry we specify `dnsZones`, which matches *all* domains in the zones specified (i.e. this is a more specific match that the other one).